dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java

```diff
@@ -865,4 +866,103 @@ private boolean isSecretDestroyed(final char [] chars){
         return true;
     }
 
+    /***
+     * Given scenario: We create a file then move it into the system folder we clear cache and the the request app-descriptors again
+     * Expected: The Key must appear marked as System-app. If we attempt a delete
+     *
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws IOException
+     * @throws URISyntaxException
+     */
+    @Test(expected = DotSecurityException.class)
+    public void Test_Add_System_File_Retrieve_Descriptors_Verify_()
+            throws DotDataException, DotSecurityException, IOException, URISyntaxException {
+            //Generate a yml file
+        final AppDescriptorDataGen dataGen = new AppDescriptorDataGen()
+                .stringParam("p1", false,  true)
+                .stringParam("p2", false,  true)
+                .stringParam("p3", false,  true)
+                .withName("system-app-example")
+                .withDescription("system-app-demo")
+                .withExtraParameters(false);
+        final File file = dataGen.nextPersistedDescriptor();
+
+        //Move the file to the system folder
+        final Path systemAppsDescriptorDirectory = AppsAPIImpl.getSystemAppsDescriptorDirectory();
+        final boolean result = file.renameTo(new File(systemAppsDescriptorDirectory.toString() + File.separator + file.getName()));
+        Assert.assertTrue(result);
+
+        final User admin = TestUserUtils.getAdminUser();
+        final AppsAPI api = APILocator.getAppsAPI();
+        final AppsCache appsCache = CacheLocator.getAppsCache();
+
+        //Invalidate cache so the new descriptors get picked
+        appsCache.invalidateDescriptorsCache();
+        final List<AppDescriptor> appDescriptors = api.getAppDescriptors(admin);
+
+        //Verify the file we just submitted is recognized as a system-app-file
+        final Optional<AppDescriptor> optional = appDescriptors.stream()
+                .filter(appDescriptor -> dataGen.getKey().equals(appDescriptor.getKey())).findFirst();
+        Assert.assertTrue(optional.isPresent());
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertTrue' due to existing static import 'org.junit.Assert.assertTrue'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5903396)